<center style="color:red;font-size:35px;padding:15px"><b> ĐỒ ÁN CUỐI KỲ </b></center>
<center style="color:blue;font-size:30px;padding:15px"><b> LẬP TRÌNH CHO KHOA HỌC DỮ LIỆU</b></center>
<center style="color:black;font-size:20px;padding:5px"><b><a href="https://github.com/dhptrung/P4DS-FinalProject">Thùng chứa GitHub</a></b></center>

---

| MSSV | Họ và tên | Tài khoản GitHub |
|:----:|:---:|:---:|
|18120066| Bùi Đoàn Hữu Nhân | https://github.com/bdhnhan |
|18120097| Đinh Hữu Phúc Trung| https://github.com/dhptrung |

# Import các thư viện cần thiết

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

# Thu thập dữ liệu

## Dữ liệu là về chủ đề gì và lấy từ nguồn nào?

- Chủ đề: Về tuổi thọ của các quốc gia, là tập hợp các yếu tố ảnh hưởng đến tuổi thọ trung bình của các quốc gia trên thế giới.
- Nguồn: Tập dữ liệu được lấy từ Kaggle - [WHO national life expectancy](https://www.kaggle.com/mmattson/who-national-life-expectancy)

## Người ta có cho phép dùng dữ liệu như này không?

- Đây là bộ dữ liệu được tác giả công khai và cung cấp theo giấy phép **CC0: Public Domain (Giấy phép Creative Commons)**.
- Với giấy phép trên tác giả từ chối toàn bộ quyền sở hữu đối với bộ dữ liệu này. Bắt cứ ai đều có thể sao chép, sửa đổi, phân phối, ngay cả cho mục đích thương mại mà không cần xin phép.

## Người ta thu thập dữ liệu như thế nào?

- Bộ dữ liệu này được WHO cung cấp và được thu thập từ nhiều nguồn, chủ yếu là sử dụng thông tin của GHO (Đài quan sát sức khỏe toàn cầu) và UNESCO (Tổ chức Giáo dục, Khoa học và Văn hóa Liên Hợp Quốc).
- Tập dữ liệu được sử dụng trong đồ án này là một phần nhỏ các yếu tố ảnh hưởng đến tuổi thọ của 183 quốc gia và trong khoảng thời gian từ 2000-2016. 

# Khám phá dữ liệu

## Đọc dữ liệu vào dataframe

In [2]:
df = pd.read_csv('who_life_exp.csv')
df.head()

country country_code  region  year  life_expect  life_exp60  \
0  Angola          AGO  Africa  2000     47.33730    14.73400   
1  Angola          AGO  Africa  2001     48.19789    14.95963   
2  Angola          AGO  Africa  2002     49.42569    15.20010   
3  Angola          AGO  Africa  2003     50.50266    15.39144   
4  Angola          AGO  Africa  2004     51.52863    15.56860   

   adult_mortality  infant_mort  age1-4mort  alcohol  ...  che_gdp    une_pop  \
0         383.5583     0.137985    0.025695  1.47439  ...  1.90860  16395.473   
1         372.3876     0.133675    0.024500  1.94025  ...  4.48352  16945.753   
2         354.5147     0.128320    0.023260  2.07512  ...  3.32946  17519.417   
3         343.2169     0.122040    0.021925  2.20275  ...  3.54797  18121.479   
4         333.8711     0.115700    0.020545  2.41274  ...  3.96720  18758.145   

   une_infant  une_life  une_hiv  une_gni  une_poverty  une_edu_spend  \
0       122.2    46.522      1.0   2530.0         32.3        2.60753   
1       118.9    47.059      1.1   2630.0          NaN            NaN   
2       115.1    47.702      1.2   3180.0          NaN            NaN   
3       110.8    48.440      1.3   3260.0          NaN            NaN   
4       106.2    49.263      1.3   3560.0          NaN            NaN   

   une_literacy  une_school  
0           NaN         NaN  
1      67.40542         NaN  
2           NaN         NaN  
3           NaN         NaN  
4           NaN         NaN  

[5 rows x 32 columns]

## Dữ liệu có bao nhiêu dòng và bao nhiêu cột?

In [3]:
df.shape

(3111, 32)

Dữ liệu có **3111 dòng** và **32 cột**.

## Mỗi dòng có ý nghĩa gì? Có vấn đề các dòng có ý nghĩa khác nhau không?

## Dữ liệu có các dòng bị lặp không?

In [6]:
have_duplicated_rows = df.duplicated().any()
have_duplicated_rows

False

Dữ liệu không có dòng nào bị lặp.

## Mỗi cột có ý nghĩa gì?

## Mỗi cột hiện đang có kiểu dữ liệu gì? Có cột nào có kiểu dữ liệu chưa phù hợp để có thể xử lý tiếp không?

In [8]:
df.dtypes

country             object
country_code        object
region              object
year                 int64
life_expect        float64
life_exp60         float64
adult_mortality    float64
infant_mort        float64
age1-4mort         float64
alcohol            float64
bmi                float64
age5-19thinness    float64
age5-19obesity     float64
hepatitis          float64
measles            float64
polio              float64
diphtheria         float64
basic_water        float64
doctors            float64
hospitals          float64
gni_capita         float64
gghe-d             float64
che_gdp            float64
une_pop            float64
une_infant         float64
une_life           float64
une_hiv            float64
une_gni            float64
une_poverty        float64
une_edu_spend      float64
une_literacy       float64
une_school         float64
dtype: object

Tất cả các cột đều đã có kiểu dữ liệu phù hợp.

# Tiền xử lý dữ liệu

# Khám phá dữ liệu (tiếp tục)

## Với mỗi cột có kiểu dữ liệu dạng số (numerical), các giá trị được phân bố như thế nào?

## Với mỗi cột có kiểu dữ liệu dạng phân loại (categorical), các giá trị được phân bố như thế nào?

# Đưa ra các câu hỏi cần trả lời

## Câu hỏi là gì?

## Nếu trả lời được câu hỏi thì sẽ có lợi ích gì?

# Tiền xử lý + phân tích dữ liệu để trả lời câu hỏi

## Câu 1

### Tiền xử lý (nếu có)

### Phân tích

## Câu 2

### Tiền xử lý (nếu có)

### Phân tích

## Câu 3

# Nhìn lại quá trình làm đồ án

## Đã gặp những khó khăn gì?

### Nhân

### Trung

## Có học được gì hữu ích không?

### Nhân 

### Trung

## Nếu có thêm thời gian thì sẽ làm gì?

# Tài liệu tham khảo